plugins/livelog/livelog.go

```diff
@@ -100,11 +103,42 @@ func (tp *taskPlugin) Finished(success bool) error {
 		return err
 	}
 	defer file.Close()
+
+	tempDir, err := runtime.NewTemporaryStorage(os.TempDir())
+	if err != nil {
+		return err
+	}
+
+	tempFile, err := tempDir.NewFile()
+	if err != nil {
+		return err
+	}
+
+	defer tempFile.Close()
+
+	zip := gzip.NewWriter(tempFile)
+	if _, err = io.Copy(zip, file); err != nil {
+		return err
+	}
+
+	if err = zip.Close(); err != nil {
+		return err
+	}
+
+	compressedLog, err := os.Open(tempFile.Path())
```

Comment:
temp file already implements `ioext.ReadWriteSeeker`

It just embeds `*os.File`, see: https://github.com/taskcluster/taskcluster-worker/blob/2619dd09aeb69107e70f5a7dc9d788d8552f6954/runtime/tempfolder.go#L41